README.md

```diff
@@ -8,4 +8,30 @@ This is a placeholder README.md for a new repository
 ### License
 
 This code is open source software licensed under the [Apache 2.0 License]("http://www.apache.org/licenses/LICENSE-2.0.html").
-    
\ No newline at end of file
+
+
+***
+
+# Instructions
+
+Tool that generates a TOTP (timed-based-one-time password) from a secret.
+It uses the HMAC-SHA512 encryption
+
+### Packaging
+
+Run `sbt clean assembly`
+
+### Running
+
+./generate-totp.sh [TOTP_SECRET]
+
+example: `./generate-totp.sh BZDHSGCIGQXRWVQX`
+
+
+***
+
+# More info:
+* [TOTP algorithm](https://en.wikipedia.org/wiki/Time-based_One-time_Password_Algorithm)
+* [TOTP algorithm specifications](https://tools.ietf.org/html/rfc6238)
+* [HOTP algorithm specifications](https://tools.ietf.org/html/rfc4226)
+* [Oauth 2.0 specifications](https://tools.ietf.org/html/rfc6749)
```

Comment:
We need to give a _lot_ more context here - for instance: 

* What use case is it built for (i.e. it's for testing purposes only - not intended to be part of a prod stack)?
* How many digits does it use? (i.e. 8)
* What is the OAuth 2.0 link necessary - we haven't mentioned that it's for testing out a flow we have as part of OAuth 2.0.